libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.cpp

```diff
@@ -282,6 +304,16 @@ HTTPUpdateResult ESP8266HTTPUpdate::handleUpdate(HTTPClient& http, const String&
         http.addHeader(F("x-ESP8266-version"), currentVersion);
     }
 
+    if (_user && _user[0] != 0x00 && _password && _password[0] != 0x00)
```

Comment:
prefer _user. isEmpty () and _password. isEmpty ()

Reply:
Actually it was a copy-paste solution to meet my needs, from the [ESP8266HTTPClient](https://github.com/esp8266/Arduino/tree/83166f948bedff85543b1cb5532b905746744df7/libraries/ESP8266HTTPClient/src) library.